### PR TITLE
fix error with hash keys beginning with '-'

### PR DESCRIPTION
**Pull request by dod38fr**
Hello

YAML::Tiny fails to parse a hash where the key begins with '-' (e.g. '-foo'). YAML::Tiny fails with:
```
YAML::Tiny found illegal characters in plain scalar: 'family: 'Courier 10 Pitch'' at test-yaml.pl line 21.
```

This PR contains a fix to this problem and a new test case for this style of hash keys.

Here's the test program that triggers this error:

```
$ cat test-yaml.pl
use strict;
use warnings;
use YAML::Tiny;
use XXX;

YYY (YAML::Tiny->read('test.yml') );
```

The yaml file:
```
$ cat test.yml
---
font:
  -family: 'Courier 10 Pitch'
  -overstrike: 0
  -size: -13
  -slant: roman
  -underline: 0
  -weight: normal
```

The result of the script with current version of YAML::Tiny:

```
$ perl test-yaml.pl
YAML::Tiny found illegal characters in plain scalar: 'family: 'Courier 10 Pitch'' at test-yaml.pl line 6.
```

And the result with the patch:
```
$ perl -Ilib test-yaml.pl
--- !!perl/array:YAML::Tiny
- font:
    -family: Courier 10 Pitch
    -overstrike: 0
    -size: -13
    -slant: roman
    -underline: 0
    -weight: normal
...
  at test-yaml.pl line 6
  ```

This patch also fixes [RT #85045](https://rt.cpan.org/Public/Bug/Display.html?id=85045)

All the best

### CODE REVIEW

**Comment by karenetheridge** [comment]
Your example is a bit hard to follow because of the use of XXX, which also includes yaml encoding/decoding inside of it.

**Comment by karenetheridge** [comment]
I suspect the problem here is not that keys starting with `-` cannot be used, but rather that they don't round-trip:  parsing works fine when the key is denoted with quotes, but dumping the yaml does not add them:

```
perl -MYAML::Tiny=Load -MData::Dumper -wle'print Dumper(Load(qq{---\nfont:\n  "-family": 1\n}))'
$VAR1 = {
          'font' => {
                      '-family' => '1'
                    }
        };

perl -Ilib -MYAML::Tiny=Dump -wle'print Dump({font => { "-family" => 1 }})'
---
font:
  -family: 1
```

We'll need to look at what YAML::XS does to be sure.

**Comment by dod38fr** [comment]
According to YAML spec:

> A block sequence is simply a series of nodes, each denoted by a leading “-” indicator. The “-” indicator must be separated from the node by white space. This allows “-” to be used as the first character in a plain scalar if followed by a non-space character (e.g. “-1”).

So I believe that the dump done by YAML::Tiny is fine: `-family: 1` is a "block mapping" :
```
$ perl -MYAML=Load -MData::Dumper -wle 'print Dumper(Load(qq{---\nfont:\n  -family: 1\n}))'
$VAR1 = {
          'font' => {
                      '-family' => '1'
                    }
        };
```

`"-family": 1` is also a "block mapping" with extra quotes:
```
$  perl -MYAML=Load -MData::Dumper -wle 'print Dumper(Load(qq{---\nfont:\n  "-family": 1\n}))'
$VAR1 = {
          'font' => {
                      '-family' => '1'
                    }
        };
```

whereas `- family: 11` is "block mapping" in a "block sequence":
```
$ perl -MYAML=Load -MData::Dumper -wle 'print Dumper(Load(qq{---\nfont:\n  - family: 1\n}))'
$VAR1 = {
          'font' => [
                      {
                        'family' => '1'
                      }
                    ]
        };
```

The patch I propose is necessary for YAML::Tiny to behave like YAML:
```
$ perl -MYAML::Tiny=Load -MData::Dumper -wle 'print Dumper(Load(qq{---\nfont:\n  -family: 1\n}))'
YAML::Tiny found illegal characters in plain scalar: 'family: 1' at -e line 1.
$ perl -Ilib -MYAML::Tiny=Load -MData::Dumper -wle 'print Dumper(Load(qq{---\nfont:\n  -family: 1\n}))'
$VAR1 = {
          'font' => {
                    '-family' => '1'
                  }
        };

$ perl -MYAML=Load -MData::Dumper -wle 'print Dumper(Load(qq{---\nfont:\n  -family: 1\n}))'
$VAR1 = {
          'font' => {
                      '-family' => '1'
                    }
        };
```

All the best

File: lib/YAML/Tiny.pm [hunk 1]
```diff
@@ -528,7 +528,7 @@ sub _load_hash {
                 $hash->{$key} = undef;
                 return 1;
             }
-            if ( $lines->[0] =~ /^(\s*)-/ ) {
+            if ( $lines->[0] =~ /^(\s*)-(?:\s|$)/ ) {
```
Comment (dolmen) [inline lib/YAML/Tiny.pm]: White space as defined in the YAML space is stricter than Perl's `\s`: YAML white space is only spaces and tab
http://www.yaml.org/spec/1.2/spec.html#id2775170

Comment (dod38fr) [reply lib/YAML/Tiny.pm]: I had completely forgotten about this PR. Sorry about that.

I've updated this PR and rebased it on `develop` branch

**Comment by karenetheridge** [comment]
Could you rebase this against master please? You should be able to reopen and adjust the merge base at the top of this page.

**Comment by dod38fr** [comment]
Hi
I've rebased my branch.

Unfortunately, I cannot re-open this PR. 

Can you re-open it on your side ?

All the best

**Comment by karenetheridge** [comment]
I cannot open it until/before the base branch has been changed (there is an option at the top of this PR page to do it) - it needs to be master, not devel.

**Comment by dod38fr** [comment]
I don't have the option to change the base branch after I click on "edit" button. I can  only change  the title of the PR.

I guess we're stuck.

Would you like me to open another PR with the right target branch ?

All the best
